pkgs/applications/networking/instant-messengers/linphone/default.nix

```diff
@@ -1,61 +1,142 @@
+    readline openldap cyrus_sasl libupnp zlib libxml2 gtk2 libnotify speex
+    ffmpeg libX11 mbedtls libsoup udev ortp mediastreamer sqlite belle-sip
+    libosip libexosip bctoolbox libmatroska gdk-pixbuf glib cairo pango bzrtp
+    belcard bcg729 liblinphone minizip2 qtbase qtquickcontrols2 qttranslations
+    qtgraphicaleffects belr
   ];
 
   nativeBuildInputs = [
     intltool pkgconfig makeWrapper cmake bcunit doxygen graphviz
-    (python.withPackages (ps: [ ps.pystache ps.six ]))
   ];
 
-  NIX_CFLAGS_COMPILE = [
-    "-Wno-error"
-    "-I${glib.dev}/include/glib-2.0"
-    "-I${glib.out}/lib/glib-2.0/include"
-    "-I${gtk2.dev}/include/gtk-2.0/"
-    "-I${cairo.dev}/include/cairo"
-    "-I${pango.dev}/include/pango-1.0"
-    "-I${gtk2}/lib/gtk-2.0/include"
-    "-DLIBLINPHONE_GIT_VERSION=\"v${version}\""
+  cmakeFlags = [
+    "-DCMAKE_BUILD_TYPE=RelWithDebInfo"
+    "-DMINIZIP_INCLUDE_DIRS=${minizip2}/include"
+    "-DMINIZIP_LIBRARIES=minizip"
   ];
 
-  postInstall = ''
-    for i in $(cd $out/bin && ls); do
-      wrapProgram $out/bin/$i --set MEDIASTREAMER_PLUGINS_DIR ${mediastreamer-openh264}/lib/mediastreamer/plugins
-    done
+  # The default install phase fails because the paths are somehow messed up in
+  # the makefiles. The errors were like:
+  #
+  #   CMake Error at cmake_builder/linphone_package/cmake_install.cmake:49 (file):
+  #     file INSTALL cannot find
+  #     "/build/linphone-desktop-.../build/linphone-sdk/desktop//nix/store/.../bin":
+  #     No such file or directory.
+  #
+  # If someone is able to figure out how to fix that, great. For now, just
+  # trying to pick all the relevant files to the output.
+  #
+  # Also, some grammar files needed to be copied too from some dependencies. I
+  # suppose if one define a dependency in such a way that its share directory is
+  # found, then this copying would be unnecessary. These missing grammar files
+  # were discovered when linphone crashed at startup and it was run with
+  # --verbose flag.
+  #
+  # It is quite likely that there are some other files still missing and
+  # Linphone will randomly crash when it tries to access those files. Then,
+  # those just need to be copied manually below.
+  installPhase = ''
+    mkdir -p $out/bin
+    cp linphone $out/bin/
+    mkdir -p $out/share/applications
+    cp linphone.desktop $out/share/applications/
+    cp -r ../assets/icons $out/share/
+    mkdir -p $out/share/belr/grammars
+    cp -r ${belcard}/share/belr/grammars/* $out/share/belr/grammars/
+    cp -r ${liblinphone}/share/belr/grammars/* $out/share/belr/grammars/
+    mkdir -p $out/share/linphone
+    cp -r ${liblinphone}/share/linphone/* $out/share/linphone/
   '';
 
-  meta = with stdenv.lib; {
+  meta = with lib; {
     homepage = https://www.linphone.org/;
     description = "Open source SIP phone for voice/video calls and instant messaging";
-    license = licenses.gpl2Plus;
+    license = licenses.gpl3;
```

Comment:
Is it GPLv3 or GPLv3+?

Reply:
GPLv3 according to this: https://gitlab.linphone.org/BC/public/linphone-desktop#license and this: https://gitlab.linphone.org/BC/public/linphone-desktop/blob/master/LICENSE.txt

Reply:
But in some source file headers it says GPLv2+ and some GPLv3+ so who knows.

Reply:
I wonder if this is worth an upstream issue…

Reply:
It's not necessarily an issue as there's no licensing conflict if there are files with GPL2+ and GPL3+ but the overall package is GPL3. But yeah, could be that they mean to license the whole package under GPL3+.